querybook/server/lib/query_analysis/templating.py

```diff
@@ -63,16 +71,67 @@ def get_default_variables():
     }
 
 
-def get_templated_variables_in_string(s: str) -> Set[str]:
+def create_get_latest_partition(engine_id) -> Callable[[str, str], str]:
+    metastore_loader = None
+    with DBSession() as session:
+        engine = admin_logic.get_query_engine_by_id(engine_id, session=session)
+        metastore_id = engine.metastore_id if engine else None
+        metastore_loader = (
```

Comment:
since now it is done in the outer function, can you also verify metastore_loader exists here? sorry for confusion in the previous comment

Reply:
oops yeah I actually moved that down since I originally made `engine_id` not required, but fixed it 👍